### PR TITLE
Laying ground for more structured OData section and adapting FAQ

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context

- [x] Adding the FAQ section in a similar way to Java. Developers would always look only into segments dedicated to their platform.
- [x] Adding Overview section to Odata, currently generic, in the future to hold feature matrix for OData and use-cases.

Please, review with suggestions when possible.

### CODE REVIEW

File: docs/js/faq.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+---
+id: frequently-asked-questions
+title:  Frequently asked questions
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: FAQ
+description: You asked! We Answered! We collected here the most frequent question about SAP Cloud SDK for Java.
```
Comment (sacnl) [inline docs/js/faq.md:7]:
```suggestion
description: You asked! We Answered! We collected here the most frequent question about SAP Cloud SDK for JavaScript.
```

File: docs/js/faq.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+### I receive an OData error/exception when using Cloud SDK for JavaScript
+
+You may see some errors while developing. These errors are not always caused by flaws in the SDK as we often see inconsistent OData protocol handling by different services. Some of them even have known flaws for which we have workarounds. If you can't solve your issue via debugging and experimenting, please, report your incident via [Stack Overflow](https://stackoverflow.com/questions/tagged/sap-cloud-sdk ) or [Github](https://github.com/SAP/cloud-sdk/issues/new/choose).
+
+## REST related questions ##
+
+### Do you support REST client libraries?
+
+We're working on it. The current support in experimental and not ready for production use.
```
Comment (sacnl) [inline docs/js/faq.md:85]:
```suggestion
We're working on it. The current support is experimental and not ready for production use.
```

File: docs/js/faq.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+## Questions about SAP Cloud Platform
+
+### Do you support SAP Cloud Platform - Cloud Foundry?
+
+SAP Cloud SDK for JavaScript has first-class support for [SCP Cloud
+Foundry](https://www.sap.com/products/cloud-platform.html). We provide plenty of helpful abstractions for [connectivity](features/connectivity/destination-js-sdk) and authentication that make developing Apps a pleasant and rewarding experience. Let us know if you're missing any features about SCP Cloud Foundry support from SAP Cloud SDK for Java.
+
+### Do you support SAP Cloud Platform - Neo?
```
Comment (sacnl) [inline docs/js/faq.md:94]: This is not as relevant for JavaScript, as Neo only supports Java runtimes for backend.

Comment (artemkovalyov) [reply docs/js/faq.md:94]: Learn one thing a day they say :) I'll remove it.

**Pull request merged**